### PR TITLE
PACA-1030: Update RDA Bridge code for RDA 0.13

### PR DESCRIPTION
**Pull request by KyRobbins**
**JIRA Ticket:**
[PACA-1030](https://jira.cms.gov/browse/PACA-1030)

**User Story or Bug Summary:**

---

### What Does This PR Do?

To better support the RDA 0.13 changes, the RDA Bridge must be updated to include the new fields so new synthetic data can be generated and added to the SandBox environment.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java [hunk 1]
```diff
@@ -50,20 +54,19 @@ public Optional<MessageOrBuilder> transform(
 
     int lineNumber = getLineNumber(data, Fiss.CLM_LINE_NUM);
 
-    if (message.getMessage() instanceof FissClaimChange) {
+    if (message.getMessage() instanceof FissClaimChange storedClaim) {
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java:57]: Nice use of a newly available feature to make the code clearer!

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java [hunk 2]
```diff
@@ -141,10 +149,10 @@ FissClaimChange transformNewClaim(
             .setCurrLoc1Unrecognized("?")
             .setCurrLoc2Unrecognized("?")
             .setCurrStatusEnum(FissClaimStatus.CLAIM_STATUS_ROUTING)
-            .setCurrTranDtCymd("1970-01-01")
+            .setCurrTranDtCymd(DEFAULT_HARDCODED_DATE)
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java:152]: Nice.

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java [hunk 3]
```diff
@@ -228,31 +239,16 @@ FissClaimChange transformNewClaim(
...
-        break;
-      case "snf":
-        indicator = FissClaimTypeIndicator.CLAIM_TYPE_SNF;
-        break;
-      default:
-        indicator = FissClaimTypeIndicator.UNRECOGNIZED;
-    }
-
-    return indicator;
+    return switch (fileName.trim()) {
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java:242]: Again, nice.

**Review by brianburton — APPROVED** [review]
Nice work.

**Review by daryll — APPROVED** [review]
lgtm

File: apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/McsTransformer.java [hunk 4]
```diff
@@ -45,10 +45,8 @@ public Optional<MessageOrBuilder> transform(
 
     int lineNumber = getLineNumber(data, Mcs.LINE_NUM);
 
-    if (message.getMessage() instanceof McsClaimChange) {
+    if (message.getMessage() instanceof McsClaimChange storedClaim) {
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/McsTransformer.java:48]: Lots of Java17 nicety in here

**Review by lsmitchell — APPROVED** [review]

**Pull request merged**